andela/knights-bn-backend
# #170947687 User should be able to like/unlike accommodation

#### What does this PR do?
- Add functionality for allowing a traveler to like or unlike a specific accommodation
#### Description of Task to be completed?
- User can like or dislike a specific accommodation
#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn   backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-like-unlike-accommodation-170947687` 
- run `npm install` to install dependencies
- check if you have PostgreSQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations
- run `npm run dropTables` to clean your database
- run `npm run createTables` to create tables in your new database you have created.
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have
- you can log in with default credentials below
```
{
    "email": "alain.maxime@gmail.com",
    "password": "Niyonkuru@1"
}
```
- view all available accommodation facilities by sending a GET request to this URL `http://localhost:4000/api/v1/view/accommodations`  in order to find an accommodation you want to like or dislike
- chose the one you want to like or dislike by grabbing its ID and append it at the end of this URL `http://localhost:4000/api/v1/accommodation/dislike/1` or this one `http://localhost:4000/api/v1/accommodation/like/1` and sent POST request to it
- to view, all the accommodations with their likes and dislikes sent GET request to this URL `http://localhost:4000/api/v1/view/accommodations`
- to view specific accommodation along with its likes and dislikes send a GET request to this URL `http://localhost:4000/api/v1/view/accommodation/1` the number one `(1)` you see here is accommodation `ID`.  you can replace it with another accommodation `ID`  you want to view

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947687](https://www.pivotaltracker.com/story/show/170947687)
#### Screenshots

#### perform like
<img width="858" alt="Screen Shot 2020-03-19 at 17 39 58" src="https://user-images.githubusercontent.com/36619897/77086349-54043300-6a0a-11ea-9149-6cb2cbb22e7b.png">
#### perform dislike
<img width="858" alt="Screen Shot 2020-03-19 at 17 39 03" src="https://user-images.githubusercontent.com/36619897/77086400-6a11f380-6a0a-11ea-92cd-821778436d9f.png">

#### view likes and dislikes of a specific accommodation 
<img width="858" alt="Screen Shot 2020-03-19 at 17 41 19" src="https://user-images.githubusercontent.com/36619897/77086563-a04f7300-6a0a-11ea-9be4-b0c25929de7d.png">
